catalog/files/impl/src/main/java/org/projectnessie/catalog/files/gcs/GcsBucketOptions.java

```diff
@@ -72,6 +73,29 @@ default GcsAuthType effectiveAuthType() {
    */
   Optional<TokenSecret> oauth2Token();
 
+  /**
+   * Flag to enable the currently experimental option to send short-lived and scoped-down
+   * credentials to clients.
+   *
+   * <p>The current default is to not enable short-lived and scoped-down credentials, but the
+   * default may change to enable in the future.
+   */
+  Optional<Boolean> downscopedCredentialsEnable();
+
+  /**
+   * The expiration margin for the scoped down OAuth2 token.
+   *
+   * <p>Defaults to the Google defaults.
+   */
+  Optional<Duration> downscopedCredentialsExpirationMargin();
+
+  /**
+   * The refresh margin for the scoped down OAuth2 token.
+   *
+   * <p>Defaults to the Google defaults.
+   */
+  Optional<Duration> downscopedCredentialsRefreshMargin();
```

Comment:
nit: we use "grace period" in other plance, e.g. in `S3Options.sessionCredentialRefreshGracePeriod()`

Reply:
Just using the Google API terminology